ors-engine/src/main/java/org/heigit/ors/export/ExportRequest.java

```diff
@@ -142,16 +180,17 @@ public ExportResult computeExport(RoutingProfile routingProfile) {
                                 extra.put("suitable", attributes.isSuitable());
                             }
                         }
-                        OsmIdGraphStorage storage2 = GraphStorageUtils.getGraphExtension(gh.getGraphHopperStorage(), OsmIdGraphStorage.class);
-                        if (storage2 != null) {
-                            extra.put("osm_id", storage2.getEdgeValue(iter.getEdge()));
+                        if (osmIdsAvailable) {
+                            extra.put("osm_id", osmIdGraphStorage.getEdgeValue(iter.getEdge()));
```

Comment:
This is duplicating information that is already present as the key in topoGeometries.
We shouldn't add it twice.

Reply:
look at my refactored ExportRequest class, should be betternow